Riku/src/Frontend/GUIFactory.cpp

```diff
@@ -194,14 +151,15 @@ CEGUI::GUI* CEGUI::GUIFactory::GetBuildingUI() {
 	return my_gui;
 }
 
-//void CEGUI::GUIFactory::UpdateResources()
-//{
-//	auto resources = front::state.getResources();
-//	CEGUI::PushButton* resButton;
-//	for (auto res : resources)
-//	{
-//		auto resButton = static_cast<CEGUI::PushButton*>(front::guiDic["GameUI"]->getWidgetByName(res.first));
-//		if (resButton != nullptr)
-//			resButton->setText(res.first + ": " + std::to_string(res.second));
-//	}
-//}
\ No newline at end of file
+CEGUI::GUI* CEGUI::GUIFactory::GetPlayerChangedUI() {
+
+	CEGUI::GUI* my_gui = new CEGUI::GUI();
+	my_gui->init();
+	my_gui->loadLayout("RikuYourTurn.layout");
+	my_gui->setFont("DejaVuSans-10");
+
+	auto onOkButton = new CEGUI::Functor::SwitchActiveGUI("GameUI", activeGUI, guiDic);
```

Comment:
To samo pytanie co wyżej

Reply:
przesunąłem stos callbacków z Fabryki do GUI i są one zwalniane w dekonstruktorze